test/integration/httpInfrastructureRest.spec.ts

```diff
@@ -0,0 +1,642 @@
+    it("should work for put204", async () => {
+      const response = await client.httpSuccess.put204();
+      assert.equal(response.status, "204");
+    });
+
+    it("should work for delete204", async () => {
+      const response = await client.httpSuccess.delete204();
+      assert.equal(response.status, "204");
+    });
+  });
+
+  describe("Failure scenarios", () => {
+    it("Should get empty should return 400", async () => {
+      const result = await client.httpFailure.getEmptyError();
+      assert.equal(result.status, "400");
+    });
+  });
+
+  describe("httpClientFailure scenarios", () => {
+    it("delete400 should return 400", async () => {
+      const result = await client.httpClientFailure.delete400();
+      assert.equal(result.status, "400");
+    });
+
+    it("patch414 should return 414", async () => {
+      const result = await client.httpClientFailure.patch414();
+      if (isNode) {
+        assert.equal(result.status, "414");
+      }
+    });
+
+    it("delete417 should return 417", async () => {
+      const result = await client.httpClientFailure.delete417();
+      assert.equal(result.status, "417");
+    });
+
+    it("get400 should return 400", async () => {
+      const result = await client.httpClientFailure.get400();
+      assert.equal(result.status, "400");
+    });
+
+    it("get402 should return 402", async () => {
+      const result = await client.httpClientFailure.get402();
+      assert.equal(result.status, "402");
+    });
+
+    it("get403 should return 403", async () => {
+      const result = await client.httpClientFailure.get403();
+      assert.equal(result.status, "403");
+    });
+
+    it("get411 should return 411", async () => {
+      const result = await client.httpClientFailure.get411();
+      assert.equal(result.status, "411");
+    });
+
+    it("get412 should return 412", async () => {
+      const result = await client.httpClientFailure.get412();
+      assert.equal(result.status, "412");
+    });
+
+    it("get416 should return 416", async () => {
+      const result = await client.httpClientFailure.get416();
+      assert.equal(result.status, "416");
+    });
+
+    it("head400 should return 400", async () => {
+      const result = await client.httpClientFailure.head400();
+      assert.equal(result.status, "400");
+    });
+
+    it("head401 should return 401", async () => {
+      const result = await client.httpClientFailure.head401();
+      assert.equal(result.status, "401");
+    });
+
+    it("head410 should return 410", async () => {
+      const result = await client.httpClientFailure.head410();
+      assert.equal(result.status, "410");
+    });
+
+    it("delete407 should return 407", async () => {
+      try {
+        const result = await client.httpClientFailure.delete407();
+      } catch (error) {
+        if (isNode) {
+          assert.equal(error.statusCode, 407);
+        } else {
+          assert.equal(error.code, "REQUEST_SEND_ERROR");
+        }
+      }
```

Comment:
how come this throws? I thought RLCs wouldn't throw based on status code

Reply:
I think we won't go into that catch step in RLC at all.

Reply:
changed